lib/icinga/checkable-check.cpp

```diff
@@ -242,6 +242,16 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 			OnReachabilityChanged(this, cr, children, origin);
 	}
 
+	if (recovery) {
+		// W/o this children could recover too late, after their problem notifications have been sent.
+		for (auto& child : children) {
+			if (child->GetProblem() && child->GetEnableActiveChecks() && !child->IsLikelyToBeCheckedSoon()) {
+				ObjectLock oLock (child);
+				child->SetNextCheck(now + 10);
```

Comment:
Why 10?

Reply:
https://github.com/Icinga/icinga2/pull/8436#discussion_r520493625

Reply:
If this was asked twice, maybe add a comment. Otherwise the next person reading the code might have the same question. :D